### PR TITLE
fix(route): linkresearcher

### LINKED ISSUE #17609: LinkResearcher无法抓取
### 路由地址

```routes
/linkresearcher/:params
```

### 完整路由地址

```fullroutes
/linkresearcher/category=theses
```

### 相关文档

https://docs.rsshub.app/zh/routes/other#link-research

### 预期是什么？

预期是显示领研的RSS

### 实际发生了什么？

TypeError: Cannot read properties of undefined (reading 'set-cookie')

### 部署

自建

### 部署相关信息

_No response_

### 额外信息

```shell
尝试了不同的自建服务，家用IP，服务器IP等，都是一样的情况。

自己访问了https://www.linkresearcher.com/theses 发现header里的确是有set-cookie的，不知道为什么代码抓不到
```

### 这不是重复的 issue

- [X] 我已经搜索了 [现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

### PR DESCRIPTION
**Pull request by KarasuShin**
## Involved Issue / 该 PR 相关 Issue

Close #17609

## Example for the Proposed Route(s) / 路由地址示例

```routes
/linkresearcher/category=theses&columns=Nature%20导读&subject=生物
/linkresearcher/category=theses&query=可视化价电子
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/routes/linkresearcher/namespace.ts [hunk 1]
```diff
@@ -2,6 +2,12 @@ import type { Namespace } from '@/types';
 
 export const namespace: Namespace = {
     name: 'Link Research',
-    url: 'linkresearcher',
+    url: 'www.linkresearcher.com',
     lang: 'en',
```
Comment (TonyRL) [inline lib/routes/linkresearcher/namespace.ts:6]:
```suggestion
    lang: 'zh-CN',
```

File: lib/routes/linkresearcher/index.ts [hunk 2]
```diff
@@ -1,76 +1,117 @@
...
-        author: item.authors === undefined ? '' : item.authors.join(', '),
-    }));
+    const items = await Promise.all(
+        pageResponse.hits.map((item) => {
+            const link = `${baseURL}/${category}/${item.id}`;
+            return cache.tryGet(link, async () => {
+                const response = await ofetch<DetailResponse>(`${apiURL}/${category === 'theses' ? 'theses' : 'information'}/${item.id}`, {
+                    responseType: 'json',
+                });
+                const dataItem: DataItem = {
```
Comment (TonyRL) [inline lib/routes/linkresearcher/index.ts:97]: Please add `doi` and `author` if it's there.

File: lib/routes/linkresearcher/index.ts [hunk 3]
```diff
@@ -1,76 +1,117 @@
...
+                    title: response.title,
+                    description: response.content,
```
Comment (TonyRL) [inline lib/routes/linkresearcher/index.ts:99]: Some theses have `enTextList` and `zhTextList` and it's suggested to use them over the `content` only

Comment (KarasuShin) [reply lib/routes/linkresearcher/index.ts:99]: You are right. I'll fix it.

File: lib/routes/linkresearcher/index.ts (hunk 2 above)
Comment (TonyRL) [reply lib/routes/linkresearcher/index.ts:97]: Yes

**Comment by cyderhu** [comment]
刚刚尝试 /linkresearcher/category=theses&columns=Nature%20导读&subject=生物
无法抓取，提示错误信息：FetchError: [POST] "https://www.linkresearcher.com/api/theses/search?from=0&size=20&type=SEARCH": <no response> fetch failed

**Pull request merged**